quantum/impl/quantum_task_queue_impl.h

```diff
@@ -24,6 +24,38 @@
 namespace Bloomberg {
 namespace quantum {
 
+namespace {
+CoroutineStateHandler wrapCoroutineStateHandler(const CoroutineStateHandler& coroutineStateHandler)
```

Comment:
Can we do 
if (not coroutineStateHandler)
   return coroutineStateHandler;
instead of wrapping an undefined coroutineStateHandler into a no-op function instead? Semantically that's the same thing, but this way we can avoid calling a no-op lambda. See the suggestion above regarding the handler call optimization.

Reply:
Done